src/java/org/apache/cassandra/index/sai/disk/format/IndexDescriptor.java

```diff
@@ -340,19 +354,18 @@ private static void deleteComponentFile(File file)
     private class IndexComponentsImpl implements IndexComponents.ForWrite
     {
         private final @Nullable IndexContext context;
-        private final Version version;
-        private final int generation;
+        private final ComponentsBuildId buildId;
 
         private final Map<IndexComponentType, IndexComponentImpl> components = new EnumMap<>(IndexComponentType.class);
 
-        // Mark groups that are complete (and should not have new components added).
-        private volatile boolean isComplete;
+        // Mark groups that are being read/have been fully written, and thus should not have new components added.
+        // This is just to catch errors where we'd try to add a component after the completion marker was written.
+        private volatile boolean sealed;
```

Comment:
is "sealed" a reserved keyword in some newer java version ?

Reply:
Yes and no. It's a "contextual reserved keyword" in java 16+ but this is not a context where it is reserved (it's only in class/interface names).